### DISCUSSION TITLE
88.0.4324.104 Flatpak Some messages diplayed at launch

### DISCUSSION THREAD

**Discussion by grandtoubab** [Eloston/ungoogled-chromium#1375] [Q&A] [answered]
**Describe the bug**
Some messages diplayed at launch:
flatpak run com.github.Eloston.UngoogledChromium
Gtk-Message: 09:42:05.207: Failed to load module "canberra-gtk-module"
Gtk-Message: 09:42:05.227: Failed to load module "canberra-gtk-module"
[39:39:0202/094208.553057:ERROR:sandbox_linux.cc(382)] InitializeSandbox() called with multiple threads in process gpu-process.

**Environment
Debian 11 Bullseye aka testing/sid
$ uname --all
Linux debian 5.10.0-3-amd64 #1 SMP Debian 5.10.12-1 (2021-01-30) x86_64 GNU/Linux
$ flatpak list
Name                                           Application ID                                    Version             Branch       Installation
ungoogled-chromium                             com.github.Eloston.UngoogledChromium              88.0.4324.104       stable       system
Codecs                                         com.github.Eloston.UngoogledChromium.Codecs                           stable       system
Freedesktop Platform                           org.freedesktop.Platform                          20.08.3             20.08        system
default                                        org.freedesktop.Platform.GL.default                                   19.08        system
default                                        org.freedesktop.Platform.GL.default                                   20.08        system
openh264                                       org.freedesktop.Platform.openh264                 2.1.0               2.0          system
GNOME Application Platform version 3.36        org.gnome.Platform                                                    3.36         system
Pitivi                                         org.pitivi.Pitivi                                 2020.09             stable       system
Codecs                                         org.pitivi.Pitivi.Codecs                                              stable       system

lspci -nnk | grep -i vga -A2
01:05.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] RS880M [Mobility Radeon HD 4225/4250] [1002:9712]
	DeviceName: 256
	Subsystem: Hewlett-Packard Company RS880M [Mobility Radeon HD 4225/4250] [103c:1443]
02:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Park [Mobility Radeon HD 5430/5450/5470] [1002:68e0]
	Subsystem: Hewlett-Packard Company Park [Mobility Radeon HD 5430/5450/5470] [103c:1443]
	Kernel driver in use: radeon

**Question
My concern is sandbox error: is it harmless or not?

**Comment by Eloston** [comment]
> My concern is sandbox error: is it harmless or not?

It happens on regular Chromium too. There is a flag that can mitigate it if you're very concerned about security, but it is at the detriment of graphics performance. I can't recall what the flag is anymore.

**Reply by BachoSeven** [reply]
The workaround to this is to export the `MESA_GLSL_CACHE_DISABLE=true` environmental variable (either in your `~/.profile` or shell configuration), see for example [this](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=918433#10) debian thread.

**Comment by grandtoubab** [comment, marked as answer]
Hello 
MESA_GLSL_CACHE_DISABLE=true 
this wokaround dont change anything.

I uses the flag  **--gpu-sandbox-start-earl**y which seems have better effect
Refer to https://peter.sh/experiments/chromium-command-line-switches/

Steps to reproduce the error

1) launch chromium in command
2) immediately close chromium using ctrl+C
3) imediately relaunch chromium using the same command with the history terminal facility ( small arrow up)

I uses the linux_portable version installed in /opt as describe in my blog:
https://bidouilledebian.wordpress.com/2021/01/18/de-google-iser-son-navigateur/

and my final command is now

/opt/ungoogled-chromium/chrome --enable-namespace-sandbox --gpu-sandbox-start-early --no-pings --disk-cache-dir=/tmp/chromium --password-store=gnome --renderer-process-limit=2 --lang=fr --disable-gpu-rasterization --disable-background-networking %U


It  remains only one error 

```
guy@debian:~$ /opt/ungoogled-chromium/chrome --enable-namespace-sandbox --gpu-sandbox-start-early --no-pings --disk-cache-dir=/tmp/chromium --password-store=gnome --renderer-process-limit=2 --lang=fr --disable-gpu-rasterization --disable-background-networking %U
[9688:9711:0204/094310.671571:ERROR:ev_root_ca_metadata.cc(841)] Failed to register OID: 0
```
